src/science/mathematics/linear_algebra/vector-norm.md

```diff
@@ -1,48 +1,58 @@
 ## Vector Norm
 [![ndarray-badge]][ndarray]
 
-This recipe demonstrates use of the [`Array1`] type, [`dot`], [`map`], and
-[`scalar_sum`] in computing the [l1] and [l2] norms of a given vector. The l2
-norm calculation is the simpler of the two, as it is the square root of the dot
-product of a vector with itself, shown in the function `l2_norm`. The l1 norm,
-shown in the function `l1_norm` is computed by mapping the elements of the input
-vector to their absolute value, and then calling `scalar_sum`, which reduces the
-elements via summation.
-
-Note that both `l1_norm` and `l2_norm` take the [`Array1`] type. This recipe
+This recipe demonstrates use of the [`Array1`] type, [`ArrayView1`] type,
+[`fold`] method, and [`dot`] method in computing the [l1] and [l2] norms of a
+given vector. The l2 norm calculation is the simpler of the two, as it is the
+square root of the dot product of a vector with itself, shown in the function
+`l2_norm`. The l1 norm, shown in the function `l1_norm`, is computed by a `fold`
+operation that sums the absolute values of the elements. (This could also be
+performed with `x.mapv(f64::abs).scalar_sum()`, but that would allocate a new
+array for the result of the `mapv`.)
+
+Note that both `l1_norm` and `l2_norm` take the [`ArrayView1`] type. This recipe
 considers vector norms, so the norm functions only need to accept one
-dimensional arrays (hence [`Array1`]).
+dimensional views (hence [`ArrayView1`]). While the functions could take a
+parameter of type `&Array1<f64>` instead, that would require the caller to have
+a reference to an owned array, which is more restrictive than just having access
+to a view (since a view can be created from any array or view, not just an owned
+array). The most convenient argument type for the caller would be
+`&ArrayBase<S, Ix1> where S: Data`, because then the caller could use `&array`
+or `&view` instead of `x.view()`. If the function is part of your public API,
+that may be a better choice for the benefit of your users, but for internal
+functions, the more concise `ArrayView1<f64>` may be preferable.
 
 ```rust
+#[macro_use(array)]
 extern crate ndarray;
 
-use ndarray::{Array, Array1};
+use ndarray::prelude::*;
```

Comment:
Although prelude is often the library's often preferred method of including types, the cookbook prefers to be explicit about it.

Reply:
Okay, I've made it explicit again.